src/index.tsx

```diff
@@ -1,44 +1,66 @@
-import React, { useEffect, useRef, useState } from 'react';
-import { Chessground as ChessgroundApi } from 'chessground';
+import React, { forwardRef, useEffect, useImperativeHandle, useRef, useState } from 'react';
 
-import { Api } from 'chessground/api';
+import { Chessground as ChessgroundApi } from 'chessground';
 import { Config } from 'chessground/config';
+import { Api } from 'chessground/api';
+import { Key } from 'chessground/types';
+
+export type LongAlgebraicMove = `${Key}${Key}`
 
 interface Props {
-  width?: number
-  height?: number
+  width?: number;
+  height?: number;
   contained?: boolean;
-  config?: Config
+  config?: Config;
 }
 
-function Chessground({
-  width = 900, height = 900, config = {}, contained = false,
-}: Props) {
-  const [api, setApi] = useState<Api | null>(null);
-
-  const ref = useRef<HTMLDivElement>(null);
-
-  useEffect(() => {
-    if (ref && ref.current && !api) {
-      const chessgroundApi = ChessgroundApi(ref.current, {
-        animation: { enabled: true, duration: 200 },
-        ...config,
-      });
-      setApi(chessgroundApi);
-    } else if (ref && ref.current && api) {
-      api.set(config);
+export interface ApiRef {
+  move: (moveStr: LongAlgebraicMove) => void;
+}
+
+const Chessground = forwardRef<ApiRef | undefined, Props>(
+  (
+    { width = 900, height = 900, config = {}, contained = false }: Props,
+    apiRef,
+  ) => {
+    const [api, setApi] = useState<Api | null>(null);
+    const divRef = useRef<HTMLDivElement>(null);
+
+    const publicApi: ApiRef = {
+      move(moveStr: LongAlgebraicMove) {
+        console.log("[MOVE]", moveStr)
+        api!.move(moveStr.substring(0, 2) as Key, moveStr.substring(2,4) as Key)
+      }
     }
-  }, [ref]);
 
-  useEffect(() => {
-    api?.set(config);
-  }, [api, config]);
+    useImperativeHandle(apiRef, () => {
+      return publicApi;
```

Comment:
What about exposing the whole chessground api instead of creating custom object?

Reply:
Sure, we can do that. Done now.